woocommerce/woocommerce
# add error messages to checkout_error event

### All Submissions:


### Changes proposed in this Pull Request:
- Pass checkout error messages / html markup to `checkout_error` event when checkout fields fail validation
- Original request [here](https://github.com/woocommerce/woocommerce/issues/25782)

Closes https://github.com/woocommerce/woocommerce/issues/25782.

### How to test the changes in this Pull Request:

1. Add a product to cart and navigate to the checkout
2. Open the browser console and enter the following:

```
jQuery( document.body ).on( 'checkout_error' , function( data , errorMessages ) {
    console.log( errorMessages );
});
```

3. Click _Place Order_ without entering any customer information in the checkout form
4. Observe that as well as the error messages appearing at the top of the checkout page, the mark up has also been logged to the console, i.e. it is now available for any other action a developer may wish to take

### Other information:

* [X] Have you successfully run tests with your changes locally? - N/A



### Changelog entry

> Enhancement - Pass error message mark up to checkout_error event
